fastestimator/trace/io/model_saver.py

```diff
@@ -31,16 +33,32 @@ class ModelSaver(Trace):
         model: A model instance compiled with fe.build.
         save_dir: Folder path into which to save the `model`.
         frequency: Model saving frequency in epoch(s).
+        max_to_keep: Maximum number of latest saved files to keep. If None, all models will be saved.
     """
-    def __init__(self, model: Union[tf.keras.Model, torch.nn.Module], save_dir: str, frequency: int = 1) -> None:
+    def __init__(self,
+                 model: Union[tf.keras.Model, torch.nn.Module],
+                 save_dir: str,
+                 frequency: int = 1,
+                 max_to_keep: Optional[int] = None) -> None:
         super().__init__(mode="train")
         self.model = model
         self.save_dir = save_dir
         self.frequency = frequency
+        self.max_to_keep = max_to_keep
+        if max_to_keep:
+            self.file_queue = queue.Queue()
 
     def on_epoch_end(self, data: Data) -> None:
         # No model will be saved when save_dir is None, which makes smoke test easier.
         if self.save_dir and self.system.epoch_idx % self.frequency == 0:
             model_name = "{}_epoch_{}".format(self.model.model_name, self.system.epoch_idx)
             model_path = save_model(self.model, self.save_dir, model_name)
             print("FastEstimator-ModelSaver: Saved model to {}".format(model_path))
+
+            if self.max_to_keep:
```

Comment:
replace all of this with:
`
rm_path = self.file_queue[self.file_queue.maxlen -1] if self.file_queue.maxlen else None
if rm_path:
  os.remove(os.path.join(self.save_dir, rm_path))
self.file_queue.appendleft(model_path)
`

Reply:
as a side note, do you need to join the save_dir with the path? doesn't save model give the full path already? if not, we probably should change the save model backend function to do that

Reply:
You are right, the os.path.join(self.save_dir, rm_path) is totally unnecessary. I will drop this.